src/roblox_api/instance.rs

```diff
@@ -235,6 +235,43 @@ impl LuaInstance {
         }
     }
 
+    fn is_a(&self, class_name: &str) -> rlua::Result<bool> {
+        let tree = self.tree.lock().unwrap();
+
+        let instance = tree
+            .get_by_ref(self.id)
+            .ok_or_else(|| rlua::Error::external("Cannot call IsA() on a destroyed instance"))?;
+
+        if class_name == "Instance" || instance.class == class_name {
+            return Ok(true);
+        }
+
+        let database = rbx_reflection_database::get();
+        let mut descriptor = database
+            .classes
+            .get(instance.class.as_str())
+            .ok_or_else(|| {
+                rlua::Error::external(format!(
+                    "Unable to obtain class {} from reflection database",
+                    &instance.class
+                ))
+            })?;
```

Comment:
Can we restructure our loop so that this line only needs to happen once? If we start with

```rust
let mut class = Some(&descriptor.class);
```

and loop from there, we should be able to only have one block that calls `classes.get`

Reply:
Let me know if that's similar to what you had in mind 🙂